framework/caching-cmm.md

```diff
@@ -149,3 +162,102 @@ the caching CMM MUST NOT store the decryption materials in the underlying CMC.
+
+### Preliminaries
+
+Each of the cache entry identifier formulas includes a serialized encryption context,
+as defined in the
+[Key Value Pairs section of the Message Header Data Format specification](../../data-format/message-header.md#key-value-pairs).
+In the following appendix sections we use `SerializeEncryptionContext`
+to denote the function that,
+given an encryption context,
+returns the serialization of the encryption context.
+
+Some of the cache entry identifier formulas include
+the two-byte algorithm suite ID for the algorithm suite in a materials request.
+The algorithm suite IDs are defined in the
+[Supported Algorithm Suites section of the Algorithm Suites specification](../../framework/algorithm-suites.md#supported-algorithm-suites).
+In the following appendix sections we use `AlgorithmSuiteId`
+to the denote the function that,
+given an algorithm suite as specified in a materials request,
+returns the corresponding two-byte algorithm suite ID.
+
+### Encryption Materials, Without Algorithm Suite
+
+If the Get Encryption Materials request does not specify an algorithm suite,
+then the cache entry identifier MUST be calculated
+as the SHA-512 hash of the concatenation of the following byte strings,
+in the order listed:
+
+1. The SHA-512 hash of the caching CMM’s Partition ID
+2. One null byte (`0x00`)
+3. The SHA-512 hash of the serialized encryption context
+
+As a formula:
+
+```
+ENTRY_ID = SHA512(
+    SHA512(cachingCMM.partitionId)
+    + 0x00
+    + SHA512(SerializeEncryptionContext(getEncryptionMaterialsRequest.encryptionContext))
+)
+```
+
+### Encryption Materials, With Algorithm Suite
+
+If the Get Encryption Materials request does specify an algorithm suite,
+then the cache entry identifier MUST be calculated
+as the SHA-512 hash of the concatenation of the following byte strings,
+in the order listed:
+
+1.  The SHA-512 hash of the caching CMM’s Partition ID
+2.  One byte with value 1 (`0x01`)
+3.  The two-byte algorithm suite ID corresponding to the algorithm suite in the request
+4.  The SHA-512 hash of the serialized encryption context
+
+As a formula:
+
+```
+ENTRY_ID = SHA512(
+    SHA512(cachingCMM.partitionId)
+    + 0x01
+    + AlgorithmSuiteId(getEncryptionMaterialsRequest.algorithmSuite)
+    + SHA512(SerializeEncryptionContext(getEncryptionMaterialsRequest.encryptionContext))
+)
+```
+
+### Decryption Materials
+
+When the caching CMM receives a Decrypt Materials request,
+it MUST calculate the cache entry identifier as
+the SHA-512 hash of the concatenation of the following byte strings,
+
+in the order listed:
+
+1.  The SHA-512 hash of the caching CMM’s Partition ID
+2.  The two-byte algorithm suite ID corresponding to the algorithm suite in the request
+3.  The concatenation of the lexicographically-sorted SHA-512 hashes of the serialized encrypted data keys,
+    where serialization is as defined in the Message Header Encrypted Data Key Entries specification.
```

Comment:
Same comment as first doc, link to spec and probably make `Message Header Encrypted Data Key Entries specification.`->`Message Header specification.`